### ISSUE TITLE
Reproducable bulids

### ISSUE THREAD

**Issue by pgerber** [vector-im/riot-android#3131]
I believe the possibility of having [reproducible builds](https://reproducible-builds.org/) should be investigated. In my opinion, it's crucial for any piece of software pushed to people's devices.

For those unfamiliar with reproducible builds, the idea is to make sure that when a binary/packages is built, everyone else can build from the same source and the result will be a binary-identical package/binary.

Advantages:
* In case of a security breach of the build infrastructure, there won't be a need to publish new binaries or even replace the app in the App store. To prove the package has not been tempered with, it can be rebuilt on arbitrary infrastructure and the resulting packages/binaries can be compared for verification.
* Builds can be done on different build infrastructure or even on ones notebook to verify the build infrastructure has not been compromised. In the blog post about a recent breach, you've not provided much detail about how the issue was discovered but you made it sound like some luck was involved. Luck is great, yet verification is better.

**Comment by eighthave** [comment]
I agree, and with the f-droid.org build, it is really close.  For example, in version code   , there was only one trivial difference: the length of the commit hash stored in _res/values/strings.xml_: 7 vs 8 chars: `<string·​name="git_revision">4​5a3aba</​string>`.  That comes from different defaults in git config.  I propose fixing this by either:

* always use the full commit ID in _strings.xml_
* always truncate the commit ID to 7 characters, separate from the git output

https://verification.f-droid.org/im.vector.alpha_70301.apk.diffoscope.html#res-values-strings.xml---res-values-strings.xml
